includes/class-signature.php

```diff
@@ -186,7 +186,14 @@ protected static function check_legacy_key_pair_for( $user_id ) {
 	 */
 	public static function generate_signature( $user_id, $http_method, $url, $date, $digest = null ) {
 		$user = Users::get_by_id( $user_id );
-		$key  = self::get_private_key_for( $user->get__id() );
+		if ( ! is_wp_error( $user ) ) {
+			$key = self::get_private_key_for( $user->get__id() );
+			$key_id = $user->get_url() . '#main-key';
+		} else {
+			$temp_sig_options = get_option( 'activitypub_temp_sig_' . $user_id );
```

Comment:
this feels a bit hacky and might break things in the future, if we maybe introduce key rotation: https://swicg.github.io/activitypub-http-signature/#key-rotation

Reply:
Do we need a sig for deletes at all? The remote server is not able to verify it anyways!?!

This is very confusing https://swicg.github.io/activitypub-http-signature/#handling-deletes-of-actors

Reply:
Agreed it is very hacky! In my previous tests Mastodon ignored actor deletes signed by the instance actor, but I will do some more tests and report back.

Reply:
or maybe we do it as you mentioned it here: https://github.com/Automattic/wordpress-activitypub/pull/552#issuecomment-1806592481

> So a workaround would be to store key pair in an option during delete_user action and delete the option during deleted_user.

Reply:
or we store the complete delete object in the schedule on the delete?!?

Reply:
> So a workaround would be to store key pair in an option during delete_user action and delete the option during deleted_user.

The first part is the hack I've implemented, the problem with the second part is that `deleted_user` will occur almost immediately, and the Delete activities will fail.

Reply:
> or we store the complete delete object in the schedule on the delete?!?

Hmm, the scheduler runs before signature generation.